### DISCUSSION TITLE
Gradients of forces with respect to particle positions

### DISCUSSION THREAD

**Discussion by sigbjobo** [deepmodeling/deepmd-kit#2419] [Q&A] [answered]
Hi, I am working on a project where I need to compute the gradient forces with respect to particle coordinates. Right now, I am thinking to do this numerically, but it is less elegant than doing it through DeePMD directly. 

Is there a simple way to achieve this with the current version of DeePMD? And if not, is there a simple way to implement this?

**Comment by wanghan-iapcm** [comment, marked as answer]
Currently it is not supported by the tf.grad. One needs to  to provides the hessian of the environment matrix, which is a customized op, with respect to the coordinates, and the corresponding ops involved in the back-propagation.
